WL-3606: Text wrong on 'ask for help' link

nicholaswilson100 opened the pull request:
There are a few things I've done.
1. Combined the 2 existing properties files into 1 and moved it to the proper Maven structure (src/main/resources).
2. Changed the pom so that it uses the new resource location.
3. Made the properties file locale-specific. (my machine picks up the en_GB.properties file)
4. Get Oxford-specific properties like 'WebLearn' from the config rather than from .properties files.
5. Lastly, change the link text on 'Ask for Help?' box in the Contact Us tool so it's the same as the third Technical one - 'Report the problem to the {0} team'.

Couple of notes:
1.   In FeedbackTool.java, there was duplicate code adding the resourcebundle as an attribute to the request which I've factored out.
Also, there were some properties in the bundle which require parameter substitution.  These are held as a constant in the class now and you can add another one here if you wanted.
2.   In FeedbackEntityProvider.java, I'm replacing an indexed parameter with the newline character because now I'm feeding a property with newline characters in it and javascript does not like them. Before we weren't putting these into javascript (the properties file was only accessed after the email was submitted).

nicholaswilson100 committed:
WL-3606: Text wrong on 'ask for help' link

There are a few things I've done.

1. Combined the 2 existing properties files into 1 and moved it to the proper Maven structure (src/main/resources).
2. Changed the pom so that it uses the new resource location.
3. Made the properties file locale-specific. (my machine picks up the en_GB.properties file)
4. Get Oxford-specific properties like 'WebLearn' from the config rather than from .properties files.
5. Lastly, change the link text on 'Ask for Help?' box in the Contact Us tool so it's the same as the third Technical one - 'Report the problem to the {0} team'.

Couple of notes:
1.   In FeedbackTool.java, there was duplicate code adding the resourcebundle as an attribute to the request which I've factored out.
Also, there were some properties in the bundle which require parameter substitution.  These are held as a constant in the class now and you can add another one here if you wanted.
2.   In FeedbackEntityProvider.java, I'm replacing an indexed parameter with the newline character because now I'm feeding a property with newline characters in it and javascript does not like them. Before we weren't putting these into javascript (the properties file was only accessed after the email was submitted).

buckett commented:
Can't we delete the unused bundle files?

nicholaswilson100 commented:
yup that's in another Jira.. but maybe it should be in this one..

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -130,4 +118,31 @@ else if (!hasViewPermission){
+    private Map<String, String> getBundle(HttpServletRequest request) {
+        Locale requestLocale = request.getLocale();
+        request.setAttribute("language", requestLocale.getLanguage());
+        ResourceBundle rb = ResourceBundle.getBundle("org.sakaiproject.feedback", requestLocale);
```
I think you probably want to use ResourceLoader all the time as it handles a users locale in the Sakai environment.

buckett commented:
Yeah, it should be in this one.

nicholaswilson100 committed:
WL-3606 : Use ResourceLoader / delete unused files

Use ResourceLoader all the time as it handles a users locale in the Sakai environment

nicholaswilson100 commented:
Cool, I've delete the unused properties files and I'm using ResourceLoader in FeedbackTool.  I've also changed the pre-existing ResourceBundle in SakaiProxy to use a ResourceLoader.

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -130,4 +119,31 @@ else if (!hasViewPermission){
     }
+
+    private Map<String, String> getBundle(HttpServletRequest request) {
+        Locale requestLocale = request.getLocale();
```
I'm not sure you should ever be getting the locale from the request. Sakai generally just uses the ResourceLoader locale all the time as there are pre-existing rules about how to work out the locale that the page should be displayed in.

nicholaswilson100 replied:
I think Adrian was putting the language of the locale in a request attribute (used in bootstrap.jsp/maybe elsewhere) rather than using it in the ResourceLoader (line 125) but it probably doesn't need to be in this private getBundle method.  I'll move it to the doGet and then getBundle doesn't need to take a HttpServletRequest as a parameter.

nicholaswilson100 replied:
If that looks okay...

nicholaswilson100 committed:
WL-3606 : Move setting locale language as request attribute out of private method

nicholaswilson100 commented:
There's a bit more coming on this one....3608,3607, 3622, 3588

nicholaswilson100 committed:
WL-3607 : 'Report a technical problem' / 'Ask for help' page has wrong text

Replacing the text for the technical instructions

nicholaswilson100 committed:
WL-3622 : change the labels of the textbox / textareas

Change the labels to "Subject" and "Summary". The error message should not say "The description must not be empty" any more. Look for other occurrences of the word 'description' and change to summary

nicholaswilson100 committed:
WL-3608 : The site contact should always be given in the drop down list of "recipient"

1.   I've removed showing the Site Contact as a label on the forms.
2.   In FeedbackTool, I'm adding the Site Contact to the list of siteUpdaters. The problem is that the list of siteUpdaters has the userId as the value of each option in dropdown whereas the site contact doesn't have to be a user.  So I'm putting the email address in there instead and testing to see if it's a valid email address the other side in FeedbackEntityProvider.

nicholaswilson100 committed:
WL-3588 : Show sensible db error to client if db error occurs

If borrowConnection or dbWrite throw a SQLexception then tell client

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -123,11 +117,48 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
+    }
+
+    private Map<String, String> getBundle(String serviceName) {
+        ResourceLoader rb = new ResourceLoader("org.sakaiproject.feedback");
```
I think you can make the ResourceLoader a static class variable as creating an ResourceLoader can be expensive.

buckett commented on src/java/org/sakaiproject/feedback/db/Database.java:
```diff
@@ -22,19 +25,19 @@ public void init() {
             if (conn != null) {
                 sqlService.returnConnection(conn);
             }
+            throw new SQLException();
```
Why not just re-throw the existing sqlException?

buckett commented on src/java/org/sakaiproject/feedback/db/Database.java:
```diff
@@ -22,19 +25,19 @@ public void init() {
-        } finally {
+        } catch (SQLException sqlException){
+            logger.error("Failed to insert feedback report. Caught sql exception while generating report. '" + FeedbackEntityProvider.DB_ERROR + "' will be returned to the client.", sqlException);
             if (conn != null) {
```
This should still be in a finally block rather than the catch.

buckett commented on src/java/org/sakaiproject/feedback/db/Database.java:
```diff
@@ -1,8 +1,11 @@
+
 import org.sakaiproject.db.api.SqlService;
 import org.apache.log4j.Logger;
+import org.sakaiproject.feedback.tool.entityproviders.FeedbackEntityProvider;
```
Is this import change needed?

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -78,32 +83,21 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 
-        request.setAttribute("enableTechnical", 
+        request.setAttribute("i18n", getBundle(serviceName));
+        request.setAttribute("language", request.getLocale().getLanguage());
```
If you want the locale you should use the method on the resource loader as it respects all the crazy Sakai rules.

buckett commented on src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:
```diff
@@ -112,12 +118,12 @@ private String handleReport(final EntityView view, final Map<String, Object> par
 
         if (title == null || title.isEmpty()) {
-			logger.error("Title incorrect. Returning " + BAD_TITLE + " ...");
+			logger.error("Subject incorrect. Returning " + BAD_TITLE + " ...");
```
These should be debug level as bad user input should log anything for the sysadmins.

nicholaswilson100 committed:
WL-3606 : various changes

rethrowing sql exception
putting connection release back in finally block
moving DB_ERROR constant
making resourceloader static
making debug level 'debug'

nicholaswilson100 commented:
Thanks.  Sorry should have seen the rethrow sql error and finally block ones..

buckett commented:
Magic. All in.
